pkg/cassandra/nodetool/client/client.go

```diff
@@ -0,0 +1,89 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+)
+
+const (
+	StorageServicePath = "read/org.apache.cassandra.db:type=StorageService"
+)
+
+type StorageService struct {
+	HostIdMap        map[string]uuid.UUID
+	LiveNodes        []string
+	UnreachableNodes []string
+	LeavingNodes     []string
+	JoiningNodes     []string
+	MovingNodes      []string
+	LocalHostId      uuid.UUID
+}
+
+type Interface interface {
+	StorageService() (*StorageService, error)
+}
+
+type client struct {
+	StorageServiceURL *url.URL
+	client            *http.Client
+}
+
+var _ Interface = &client{}
+
+func New(baseURL *url.URL, c *http.Client) *client {
+	storageServiceURL, err := url.Parse(StorageServicePath)
+	if err != nil {
+		panic(err)
```

Comment:
Better to return an error rather then panic - we can then deal with that error accordingly 'upstream'

Reply:
I'll leave this. It's parsing a string constant defined above so the error will only ever be seen by us. Similar to `resource.MustParse`: https://godoc.org/k8s.io/apimachinery/pkg/api/resource#MustParse

Reply:
Ah - did not notice that. Can we move this into an `init` function perhaps, and make `storageServiceURL` a package scoped var? These execute at the time the package is initially loaded, and there can be as many of them as you want. All you should need to do is add:

```
var storageServiceURL *url.URL
func init() {
    var err error
    storageServiceURL, err = url.Parse(StorageServicePath)
    if err != nil {
        panic(err)
    }
}
```

It'll mean the error occurs at the time the program starts, instead of further along in execution (making it a lot more obvious there's a problem with the string, as the binary itself won't start at all with it). Also saves parsing the URL multiple times 😄